lib/galaxy/managers/groups.py

```diff
@@ -0,0 +1,102 @@
+from typing import (
+    Any,
+    Dict,
+)
+
+from sqlalchemy import false
+
+from galaxy import model
+from galaxy.app import StructuredApp
+from galaxy.exceptions import (
+    Conflict,
+    ObjectAttributeMissingException,
+    ObjectNotFound,
+)
+from galaxy.managers.base import decode_id
+from galaxy.managers.context import ProvidesAppContext
+from galaxy.schema.fields import EncodedDatabaseIdField
+from galaxy.web import url_for
+
+
+class GroupsManager:
+    """Interface/service object shared by controllers for interacting with groups."""
+
+    def __init__(self, app: StructuredApp) -> None:
+        self._app = app
+
+    def index(self, trans: ProvidesAppContext):
+        """
+        Displays a collection (list) of groups.
+        """
+        rval = []
+        for group in trans.sa_session.query(model.Group).filter(model.Group.table.c.deleted == false()):
+            item = group.to_dict(value_mapper={'id': trans.security.encode_id})
+            encoded_id = trans.security.encode_id(group.id)
+            item['url'] = url_for('group', id=encoded_id)
+            rval.append(item)
+        return rval
+
+    def create(self, trans: ProvidesAppContext, payload: Dict[str, Any]):
+        """
+        Creates a new group.
+        """
+        name = payload.get('name', None)
+        if name is None:
+            raise ObjectAttributeMissingException("Missing required name")
+        self._check_duplicated_group_name(trans, name)
+
+        group = model.Group(name=name)
+        trans.sa_session.add(group)
+        user_ids = payload.get('user_ids', [])
+        users = [trans.sa_session.query(model.User).get(self._decode_id(i)) for i in user_ids]
```

Comment:
It'll be better to first decode the user ids and then to fetch them all in one query.
users = trans.sa_session.query(model.User).filter(model.User.id.in_(user_ids)).all()
The same applies to the roles.
Ideally we'd also create models for Groups and Roles.

Reply:
Done! Thanks Marius for the hint!
I was avoiding touching the database stuff but I need to learn SQLAlchemy at some point 😆